tsdb/shard.go

```diff
@@ -615,9 +615,12 @@ func (s *Shard) validateSeriesAndFields(points []models.Point, tracker StatsTrac
 	var droppedKeys [][]byte
 	if err := engine.CreateSeriesListIfNotExists(keys, names, tagsSlice, tracker); err != nil {
 		switch err := err.(type) {
-		// TODO(jmw): why is this a *PartialWriteError when everything else is not a pointer?
-		// Maybe we can just change it to be consistent if we change it also in all
-		// the places that construct it.
+		// (DSB) This was previously *PartialWriteError. Now catch pointer and value types.
+		case PartialWriteError:
```

Comment:
Should we have a test that shows this? I think both of the existing tests were already returning the right error.

Reply:
Both previous code and new code return the same error string, but new code actually completes the partial write as much as possible, instead of turning into a complete failure.  I will look to see if there's a reasonable way to verify the partial success in actual data written.